### PR TITLE
Channel info delete

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
This PR allows a user to delete a channel from the channel_info view and handles
state changes upon channel deletion.

#### Ticket Link
https://github.com/mattermost/mattermost-mobile/issues/199

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to the drivers
- [x] Has UI changes
- [ ] Includes text changes and localization file updates

### CODE REVIEW

File: service/actions/channels.js [hunk 1]
```diff
@@ -381,6 +381,11 @@ export function deleteChannel(teamId, channelId) {
             return;
         }
```
Comment (mfpiccolo) [inline service/actions/channels.js]: I was under the impression that we want to keep all view related logic out of the service actions.  In the case that  the web-hooks are not working we will need this before delete action to set the current channel.  Open to ideas here if this is not desired.

File: service/actions/channels.js [hunk 2]
```diff
@@ -381,6 +381,11 @@ export function deleteChannel(teamId, channelId) {
...
 
+        if (beforeDelete) {
+            const {action, args, _dispatch, _state} = beforeDelete;
```
Comment (enahum) [inline service/actions/channels.js]: as far as I can tell `beforeDelete` is a boolean as of the parameters in here https://github.com/mattermost/mattermost-mobile/pull/222/files#diff-21a00938e5c0137fd9703232f07fa851R130

The other thing that I might point out here is that this PR https://github.com/mattermost/mattermost-mobile/pull/221 has an Event Emitter that could be used here to emit the event and it can be subscribe elsewhere to take the action

File: app/scenes/channel_info/channel_info.js [hunk 3]
```diff
@@ -71,14 +73,23 @@ class ChannelInfo extends PureComponent {
         if (isFavorite !== this.state.isFavorite) {
             this.setState({isFavorite});
         }
-        this.navigateAfterLeave(nextProps.leaveChannelRequest);
+        this.navigateAfterLeave(nextProps.leaveChannelRequest, nextProps.deleteChannelRequest);
     }
 
-    navigateAfterLeave(leaveChannelRequest) {
-        if (
+    deleteOrLeaveRequestSuccessful(leaveChannelRequest, deleteChannelRequest) {
```
Comment (jarredwitt) [inline app/scenes/channel_info/channel_info.js]: Could this also be done in the action/view where deleteChannel and leaveChannel is fired? Looks like the deleteChannel and leaveChannel in the service/actions dispatch on failed so could you watch the request status there instead of in the view?

File: service/actions/channels.js (hunk 2 above)
Comment (jarredwitt) [inline service/actions/channels.js]: I like the idea of using the event emitter here, seems like it would handle this situation nicely.

File: app/scenes/channel_info/channel_info.js [hunk 4]
```diff
@@ -91,18 +102,19 @@ class ChannelInfo extends PureComponent {
...
     }
 
-    handleLeave() {
+    handleDeleteOrLeave(eventType) {
         const {formatMessage} = this.props.intl;
         const channel = this.props.currentChannel;
         const term = channel.type === Constants.OPEN_CHANNEL ?
             formatMessage({id: 'mobile.channel_info.publicChannel', defaultMessage: 'Public Channel'}) :
             formatMessage({id: 'mobile.channel_info.privateChannel', defaultMessage: 'Private Channel'});
+        const titleCaseEventType = `${eventType.charAt(0).toUpperCase()}${eventType.slice(1)}`;
```
Comment (hmhealey) [inline app/scenes/channel_info/channel_info.js]: Could you change this to just set the message through an if statement to keep it much cleaner? Using `formatMessage` on a formatted string is hard to follow here. I'd prefer it to be something like
```
let title;
let message;
if (eventType === 'leave') {
    title = {id: 'mobile.channel_info.alertTitleLeaveChannel', defaultMessage: 'Leave {term}'};
    message = {id: ..., defaultMessage: ...};
} else if (eventType === 'delete') {
    ...
}
```

Either that or completely separate it into `handleDelete` and `handleLeave` since there's some duplicated code, but not a ton?

File: service/actions/channels.js (hunk 2 above)
Comment (hmhealey) [inline service/actions/channels.js]: Do we even need that? Why couldn't this just use a thunk since it fires after the request has completed?

File: app/actions/views/channel.js [hunk 5]
```diff
@@ -195,10 +196,21 @@ export function unmarkFavorite(channelId) {
...
+        if (channel.isCurrent || reset) {
+            await selectInitialChannel(teamId)(dispatch, getState);
+        }
+        if (reset) {
+            dispatch({type: NavigationTypes.NAVIGATION_POP}, getState);
+        }
+    };
+}
+
+export function deleteChannel(teamId, channelId, shouldGoBack = false) {
```
Comment (hmhealey) [inline app/actions/views/channel.js]: Since we're only using this on one place, I don't think we need a new action here. You can just have
```
this.props.actions.deleteChannel(teamId, channelId).then(this.props.actions.goBack)
```
instead adding this new function with a flag that is currently always true.
